docs/source/about_mapstyle_vs_iterable.mdx

```diff
@@ -0,0 +1,220 @@
+# Differences between Dataset and IterableDataset
+
+## Downloading and streaming
+
+When you have a regular [`Dataset`], you can access it using `my_dataset[0]`: we have what we call "random access" to the rows.
+Such datasets are also called "map-style" datasets.
+For example you can download ImageNet-1k like this and access any row:
+
+```python
+from datasets import load_dataset
+
+imagenet = load_dataset("imagenet-1k", split="train")  # downloads the full dataset
+print(imagenet[0])
+```
+
+But one caveat is that you must have all the dataset stored on your disk or in memory, which makes it not always suitable for datasets bigger than disk.
+Because it can become inconvenient for big datasets, there exists another type of dataset: [`IterableDataset`].
+When you have an iterable dataset, you can access it using a `for` loop to load the data progressively as you iterate over the dataset.
+This way, only a small fraction of examples is loaded in memory, and you don't write anything on disk.
```

Comment:
```suggestion
But one caveat is that you must have the entire dataset stored on your disk or in memory, which blocks you from accessing datasets bigger than the disk.
Because it can become inconvenient for big datasets, there exists another type of dataset, the [`IterableDataset`].
When you have an `IterableDataset`, you can access it using a `for` loop to load the data progressively as you iterate over the dataset.
This way, only a small fraction of examples is loaded in memory, and you don't write anything on disk.
```